src/renderer/stores/sequencer.ts

```diff
@@ -2,36 +2,70 @@ import { v4 as uuidv4 } from "uuid";
 import { create } from "zustand";
 import { immer } from "zustand/middleware/immer";
 import {
+  BackendGlobalState,
+  CycleConfig,
   MsgState,
+  StatusType,
   TestRootNode,
+  TestSequenceContainer,
   TestSequenceElement,
+  UploadInfo,
 } from "@/renderer/types/test-sequencer";
 import { TestSequencerProject } from "@/renderer/types/test-sequencer";
+import { testSequenceRunRequest } from "@/renderer/routes/test_sequencer_panel/models/models";
+import { toast } from "sonner";
+import { createTestSequenceTree } from "@/renderer//hooks/useTestSequencerState";
+import { SendJsonMessage } from "react-use-websocket/dist/lib/types";
 
 type State = {
-  curRun: string[];
   websocketId: string;
   elements: TestSequenceElement[];
   isLocked: boolean;
   isLoading: boolean;
-  backendState: MsgState;
+  backendGlobalState: BackendGlobalState;
+  cycleConfig: CycleConfig;
+  cycleRuns: TestSequenceContainer[][];
+  playPauseState: MsgState;
   testSequenceUnsaved: boolean;
-  testSequenceTree: TestRootNode;
-  testSequencerProject: TestSequencerProject | null;
+  testSequenceStepTree: TestRootNode;
+  testSequenceDisplayed: TestSequencerProject | null;
+  sequences: TestSequenceContainer[];
+  uploadAfterRun: boolean;
+  uploadInfo: UploadInfo;
 };
 
 type Actions = {
-  markTestAsDone: (val: string) => void;
-  addTestToRunning: (val: string) => void;
-
   setWebsocketId: (val: string) => void;
   setElements: (val: TestSequenceElement[]) => void;
   setIsLocked: (val: boolean) => void;
   setIsLoading: (val: boolean) => void;
+  setBackendGlobalState: (val: BackendGlobalState) => void;
   setBackendState: (val: MsgState) => void;
   setTestSequenceUnsaved: (val: boolean) => void;
   setTestSequenceTree: (val: TestRootNode) => void;
-  setTestSequencerProject: (val: TestSequencerProject | null) => void;
+  setUploadAfterRun: (val: boolean) => void;
+  setSerialNumber: (val: string) => void;
+  setIntegrity: (val: boolean) => void;
+  setStationId: (val: string) => void;
+  setIsUploaded: (val: boolean) => void;
+  // Cycles
+  setCycleCount: (val: number) => void;
+  setInfinite: (val: boolean) => void;
```

Comment:
exposing this many setters as actions does not feel right...
are you sure you can't extract some of the mutations in the components into some composite actions that are easier to understand?
for example, if you find yourself doing `setSequences([])` often, probably just expose something like `clearSequences` instead.

Reply:
I know, but it's kind of hard to do without them. Some are necessary for things like the draggable row (like `setSequences(...)`) for example, and others are there to parameterize execution.